internal/distributed/streaming/wal.go

```diff
@@ -0,0 +1,121 @@
+package streaming
+
+import (
+	"context"
+	"sync"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+
+	"github.com/milvus-io/milvus/internal/distributed/streaming/internal/consumer"
+	"github.com/milvus-io/milvus/internal/distributed/streaming/internal/producer"
+	"github.com/milvus-io/milvus/internal/streamingcoord/client"
+	"github.com/milvus-io/milvus/internal/streamingnode/client/handler"
+	"github.com/milvus-io/milvus/internal/util/streamingutil/status"
+	"github.com/milvus-io/milvus/pkg/streaming/util/message"
+	"github.com/milvus-io/milvus/pkg/streaming/util/options"
+	"github.com/milvus-io/milvus/pkg/util/funcutil"
+	"github.com/milvus-io/milvus/pkg/util/lifetime"
+)
+
+// newWALAccesser creates a new wal accesser.
+func newWALAccesser(c *clientv3.Client) *walAccesserImpl {
+	// Create a new streaming coord client.
+	streamingCoordClient := client.NewClient(c)
+	// Create a new streamingnode handler client.
+	handlerClient := handler.NewHandlerClient(streamingCoordClient.Assignment())
+	return &walAccesserImpl{
+		lifetime:                       lifetime.NewLifetime(lifetime.Working),
+		streamingCoordAssignmentClient: streamingCoordClient,
+		handlerClient:                  handlerClient,
+		producerMutex:                  sync.Mutex{},
+		producers:                      make(map[string]*producer.ResumableProducer),
+	}
+}
+
+// walAccesserImpl is the implementation of WALAccesser.
+type walAccesserImpl struct {
+	lifetime lifetime.Lifetime[lifetime.State]
+
+	// All services
+	streamingCoordAssignmentClient client.Client
+	handlerClient                  handler.HandlerClient
+
+	producerMutex sync.Mutex
+	producers     map[string]*producer.ResumableProducer
```

Comment:
We can implement a basic eviction policy that triggers eviction if the latest access time exceeds 15 or 30 minutes to prevent potential leaks.